core/src/main/scala/kafka/cluster/Partition.scala

```diff
@@ -388,7 +388,7 @@ class Partition(val topic: String,
     val leaderHWIncremented = inWriteLock(leaderIsrUpdateLock) {
       leaderReplicaIfLocal match {
         case Some(leaderReplica) =>
-          val outOfSyncReplicas = getOutOfSyncReplicas(leaderReplica, replicaMaxLagTimeMs)
+          val outOfSyncReplicas = getOutOfSyncReplicas(leaderReplica, replicaMaxLagTimeMs).filter(r => !r.ignoreIsrCheck)
```

Comment:
@junrao 
Without the fix, running the test against current trunk fails. 
The UnderReplicatedPartitionCount value briefly increases when creating topics.

I can see the code has changed since 0.10.1. In 0.10.1, as you mentioned, lastCaughtUpTime is initialized to current time whereas in trunk it's initialized to 0. See https://github.com/apache/kafka/commit/fc88bac66686ef963b5aec53a3eba717336e4ba7#diff-66ab822b4a165d4717324669c01ba2fb

I ran the test against 0.10.1 and it's not hitting the UnderReplicatedPartitionCount issue so it looks like https://github.com/apache/kafka/commit/fc88bac66686ef963b5aec53a3eba717336e4ba7 reintroduced the issue !